script.rb

```diff
@@ -0,0 +1,184 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    length.times do
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    length.times do
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    i = 0
+    my_new_arr = [] # It will return a new array
+    length.times do
+      value = yield self[i]
+      my_new_arr.push self[i] unless value == true
+      i += 1
+    end
+    my_new_arr
+  end
+
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
+
+    i = 0
+    length.times do
+      value = yield self[i]
+      return false if value == false
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    i = 0
+    length.times do
+      value = yield self[i]
+      return true if value == true
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    i = 0
+    length.times do
+      value = yield self[i]
+      return false if value == true
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count(req = nil)
+    counter = 0
+    if req
+      my_each { |n| counter += 1 if n == req }
+    elsif block_given?
+      i = 0
+      length.times do
+        value = yield self[i]
+        counter += 1 if value == true
+        i += 1
+      end
+    else
+      to_enum(:my_count)
+    end
+    counter
+  end
+
+  def my_map(my_proc = nil)
+    return to_enum(:my_map) if my_proc.nil? && block_given?.nil?
+
+    my_new_arr = []
+    if my_proc
+      my_each { |i| my_new_arr << my_proc.call(i) } # It is applyed on itself, so it is not necessary "self.my_each"
+    else
+      my_each { |i| my_new_arr << yield(i) } # It is applyed on itself, so it is not necessary "self.my_each"
+    end
+    my_new_arr
+  end
+
+  def my_inject(init = self[0])
+    return to_enum(:my_inject) unless block_given?
+
+    index = 1
+    while index < length
+      init = yield(init, self[index])
+      index += 1
+    end
+    init
+  end
+end
```

Comment:
- [ ] Same comment as above. They all do not work as they are expected. study the original methods and implement yours to work exactly like the original.